lib/execution_segment.go

```diff
@@ -394,13 +395,151 @@ func (ess ExecutionSegmentSequence) String() string {
 	return strings.Join(result, ",")
 }
 
+// lowest common denominator
+// https://en.wikipedia.org/wiki/Least_common_multiple#Using_the_greatest_common_divisor
+func (ess ExecutionSegmentSequence) lcd() int64 {
+	var acc = ess[0].length.Denom().Int64()
+	var n int64
+	for _, seg := range ess[1:] {
+		n = seg.length.Denom().Int64()
+		if acc == n || acc%n == 0 { // short circuit
+			continue
+		}
+		acc = acc * n / gcd(acc, n)
```

Comment:
Which one specifically are you talking about :D.
 
As I wrote in the commit https://github.com/loadimpact/k6/pull/1323/commits/af4321099e8f65f9279e92c8ec46a6f7d3c1cc29 , there are better algorithms but IMO the performance difference won't be enough for the added complexity of the algorithm. 
We can always implement something better, for example [big.Int actually uses the Lehmer's algorithm for the GCD](https://github.com/golang/go/blob/master/src/math/big/int.go#L646), but that is MUCH more complex so ...

Reply:
I meant the second approach described in https://en.wikipedia.org/wiki/Least_common_multiple#Using_the_greatest_common_divisor:

> Because gcd(a, b) is a divisor of both a and b, it is more efficient to compute the LCM by dividing before multiplying. [...] This reduces the size of one input for both the division and the multiplication, and reduces the required storage needed for intermediate results (overflow in the a×b computation).

It seems like a small change, but I'm not sure if it would have performance benefits.

Reply:
I doubt the performance difference .. we still get 1 multiplication and 1 division I highly doubt the input being smaller or bigger matters for the CPU ;). But the overflow prevention is probably worth it, even if with int64 overflowing will mean some pretty huge LCD at the end ...